Visual changes and dependabot

Gijsdeman opened the pull request:
# Description
Fixes small visual issues, like "READY FOR PICKUP" text being too small for phones, and the bounce effect of orders being too big, causing them to touch/overlap when multiple spawn at once. Also wraps the Hubble logo in `<a/>` such that it redirects back to `https://hubble.cafe` on click.

It also enables dependabot, and assigns @PimVanLeeuwen automatically to the pull requests it creates to maintain the project for as long as needed. This effect already took place, as it makes PRs to the develop branch instead of main.

## Types of changes
- Bug fix _(non-breaking change which fixes an issue)_
- CI/CD _(Changes to the CI/CD configuration)_

Gijsdeman committed:
ci: dependabot

Gijsdeman committed:
fix: bounce scale and clickable logo

dependabot[bot] committed:
build(deps-dev): bump the development_minors_and_patches group across 1 directory with 3 updates

Bumps the development_minors_and_patches group with 3 updates in the /frontend directory: [@vitejs/plugin-react-swc](https://github.com/vitejs/vite-plugin-react-swc), [eslint](https://github.com/eslint/eslint) and [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite).


Updates `@vitejs/plugin-react-swc` from 3.8.0 to 3.8.1
- [Release notes](https://github.com/vitejs/vite-plugin-react-swc/releases)
- [Changelog](https://github.com/vitejs/vite-plugin-react-swc/blob/main/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite-plugin-react-swc/compare/v3.8.0...v3.8.1)

Updates `eslint` from 9.22.0 to 9.23.0
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v9.22.0...v9.23.0)

Updates `vite` from 6.2.2 to 6.2.3
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v6.2.3/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v6.2.3/packages/vite)

---
updated-dependencies:
- dependency-name: "@vitejs/plugin-react-swc"
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: development_minors_and_patches
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: development_minors_and_patches
- dependency-name: vite
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: development_minors_and_patches
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump the production_patches group across 1 directory with 3 updates

Bumps the production_patches group with 3 updates in the /frontend directory: [@gewis/tsconfig](https://github.com/GEWIS/js-configs/tree/HEAD/configs/tsconfig), [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) and [@types/react](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react).

Updates `@gewis/tsconfig` from 2.3.0 to 2.3.1
- [Release notes](https://github.com/GEWIS/js-configs/releases)
- [Changelog](https://github.com/GEWIS/js-configs/blob/main/.releaserc.json)
- [Commits](https://github.com/GEWIS/js-configs/commits/v2.3.1/configs/tsconfig)

Updates `@types/node` from 22.13.10 to 22.13.13
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

Updates `@types/react` from 19.0.10 to 19.0.12
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react)

---
updated-dependencies:
- dependency-name: "@gewis/tsconfig"
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production_patches
- dependency-name: "@types/node"
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production_patches
- dependency-name: "@types/react"
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production_patches
...

Signed-off-by: dependabot[bot] <support@github.com>

Yoronex commented on .github/dependabot.yaml:
```diff
@@ -0,0 +1,24 @@
+    target-branch: "develop"
+    assignees:
+      - "PimVanLeeuwen"
+    ignore:
+      - dependency-name: "*"
+        update-types: ["version-update:semver-major"]
```
Question: why ignore major updates? I assume those are `x` in `x.y.z`, correct? I think it is still relevant that we are at least notified of a new version via dependabot, because how else would we know?

Gijsdeman replied:
Majors require refactors and committing to a dependabot PR is a bit of a hassle. More generally this is the config we have been using / pushing to most GEWIS repos so far.

Yoronex commented on .github/dependabot.yaml:
```diff
@@ -0,0 +1,24 @@
+updates:
+  - package-ecosystem: "npm"
+    directories:
+      - "frontend"
```
Also possible for the backend (`requiements.txt`)?

Gijsdeman replied:
Did not think about that, I will add it for `pip` as well.

Yoronex commented on frontend/tsconfig.node.json:
```diff
@@ -4,6 +4,7 @@
   "compilerOptions": {
     "lib": ["DOM"],
     "types": ["node"],
+    "module": "esnext",
```
I am not a fan of using `esnext` as target, because old, unsupported devices might not be able to use Hubble Food Tracker anymore. Example: Bartjan's iPad was/is unable to open SudoPOS because the ES version was too modern and unsupported by his device.

Gijsdeman replied:
~I can put it on `es2023` which has full compatibility with all the latest browser versions (except IE, which I do not care much to support).~ See https://compat-table.github.io/compat-table/es2016plus/.

Correction, `module` only defines the output module resolution, `esnext` is plenty supported by `node 22`. The `target` is the one you are worried about, which is set to `es2022` by default in the [template config](https://github.com/GEWIS/js-configs/blob/main/configs/tsconfig/tsconfig.json). As you can find in the link above, this is supported by most browsers on their latest version.

dependabot[bot] committed:
build(deps): bump pycryptodome from 3.21.0 to 3.22.0 in /receiver

Bumps [pycryptodome](https://github.com/Legrandin/pycryptodome) from 3.21.0 to 3.22.0.
- [Release notes](https://github.com/Legrandin/pycryptodome/releases)
- [Changelog](https://github.com/Legrandin/pycryptodome/blob/master/Changelog.rst)
- [Commits](https://github.com/Legrandin/pycryptodome/compare/v3.21.0...v3.22.0)

---
updated-dependencies:
- dependency-name: pycryptodome
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump the production_patches group across 1 directory with 2 updates

Bumps the production_patches group with 2 updates in the /receiver directory: [fastapi](https://github.com/fastapi/fastapi) and [ruff](https://github.com/astral-sh/ruff).


Updates `fastapi` from 0.115.11 to 0.115.12
- [Release notes](https://github.com/fastapi/fastapi/releases)
- [Commits](https://github.com/fastapi/fastapi/compare/0.115.11...0.115.12)

Updates `ruff` from 0.11.0 to 0.11.2
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/0.11.0...0.11.2)

---
updated-dependencies:
- dependency-name: fastapi
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production_patches
- dependency-name: ruff
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: production_patches
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump framer-motion from 12.5.0 to 12.6.2 in /frontend

Bumps [framer-motion](https://github.com/motiondivision/motion) from 12.5.0 to 12.6.2.
- [Changelog](https://github.com/motiondivision/motion/blob/main/CHANGELOG.md)
- [Commits](https://github.com/motiondivision/motion/compare/v12.5.0...v12.6.2)

---
updated-dependencies:
- dependency-name: framer-motion
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
